course4/homework-string/string.hh

```diff
@@ -1,10 +1,110 @@
 #pragma once
 
+#include <cstring>
+#include <memory>
+
 namespace course {
 
 class string
 {
-	string();
+public:
+	string() : string("") {}
+	string(const char* init_str) : my_ptr(new char[std::strlen(init_str)+1])
+	{
+		std::strcpy(my_ptr.get(), init_str);
+	}
+	string(const string& other_string) : my_ptr(new char[other_string.size()+1])
+	{
+		std::strcpy(my_ptr.get(), other_string.my_ptr.get());
+	}
+	string(const char* init_str1, const char* init_str2) : my_ptr(new char[std::strlen(init_str1)+std::strlen(init_str2)+1])
+	{
+		std::strcpy(my_ptr.get(), init_str1);
+		std::strcat(my_ptr.get(), init_str2);
+	}
+
+	bool empty() const
+	{
+		return size()==0;
```

Comment:
What is the algorithmic complexity of this implementation of ```empty``` for non-empty string?
(note: this string implementation is never actually empty)